src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +70,210 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
+    // else we don't need to override the manifest. nothing to return
+}
+
+// tslint:disable:no-trailing-whitespace
+const makeLambdaDebugFile = async (params: {
+    handlerName: string,
+    debugPort: number,
+    outputDir: string
+}): Promise<{outFilePath: string, debugHandlerName: string}> => {
+    if (!params.outputDir) {
+        throw new Error('Must specify outputDir')
+    }
+    const logger = getLogger()
+
+    const [handlerFilePrefix, handlerFunctionName] = params.handlerName.split('.')
+    const debugHandlerFileName = `${handlerFilePrefix}_debug`
+    const debugHandlerFunctionName = 'lambda_handler'
+    try {
+        logger.info('pythonCodeLensProvider.makeLambdaDebugFile params:', JSON.stringify(params, undefined, 2))
+        const template = `
+import ptvsd
+from ${handlerFilePrefix} import ${handlerFunctionName} as _handler
+
+
+def ${debugHandlerFunctionName}(event, context):
+    print('waiting for debugger to attach...')
+    ptvsd.enable_attach(address=('0.0.0.0', ${params.debugPort}), redirect_output=True)
+    ptvsd.wait_for_attach()
+    print('debugger attached')
+    return _handler(event, context)
+
+`
+
+        const outFilePath = path.join(params.outputDir, `${debugHandlerFileName}.py`)
+        logger.info('pythonCodeLensProvider.makeLambdaDebugFile outFilePath:', outFilePath)
+        await writeFile(outFilePath, template)
+
+        return {
+            outFilePath,
+            debugHandlerName: `${debugHandlerFileName}.${debugHandlerFunctionName}`
+        }
+    } catch (err) {
+        logger.error('makeLambdaDebugFile failed:', err)
+        throw err
+    }
+}
+
+const fixFilePathCapitalization = (filePath: string): string => {
+    if (process.platform === 'win32') {
+        const startsWithLowercase = new RegExp(/^[a-z].*/)
+        if (startsWithLowercase.test(filePath)) {
+            return  filePath.slice(0, 1).toUpperCase() + filePath.slice(1)
+        }
+    }
+
+    return filePath
+}
+
+const makeDebugConfig = ({debugPort, samProjectCodeRoot}: {
+    debugPort?: number,
+    samProjectCodeRoot: string,
+}): PythonDebugConfiguration => {
+    return {
+        type: PYTHON_LANGUAGE,
+        request: 'attach',
+        name: 'SamLocalDebug',
+        host: 'localhost',
+        port: debugPort!,
+        pathMappings: [
+            {
+                // tslint:disable-next-line:no-invalid-template-strings
+                localRoot: fixFilePathCapitalization(samProjectCodeRoot),
+                remoteRoot: '/var/task'
+            }
+        ],
+    }
+}
+
+export async function initialize({
     configuration,
     toolkitOutputChannel,
     processInvoker = new DefaultSamCliProcessInvoker(),
     taskInvoker = new DefaultSamCliTaskInvoker()
-}: CodeLensProviderParams): void {
+}: CodeLensProviderParams): Promise<void> {
     const logger = getLogger()
 
-    const runtime = 'python3.7' // TODO: Remove hard coded value
+    const runtime = 'python3.6' // TODO: Remove hard coded value
```

Comment:
This needs to be updated before merge.

Reply:
We have a similar hardcoding for node at the moment. We do have access to the template files now, perhaps a separate backlog item + PR should be spun up to resolve the proper runtime.